networkx/classes/function.py

```diff
@@ -969,14 +969,49 @@ def all_neighbors(graph, node):
     ----------
     graph : NetworkX graph
         Graph to find neighbors.
-
     node : node
         The node whose neighbors will be returned.
 
     Returns
     -------
     neighbors : iterator
         Iterator of neighbors
+
+    Raises
+    ------
+    NetworkXError
+        If the node is not in the graph.
+
+    Examples
+    --------
+    >>> G = nx.path_graph(4)  # or DiGraph, MultiGraph, MultiDiGraph, etc
+    >>> list(nx.all_neighbors(G, 1))
+    [0, 2]
+    >>> DG = nx.DiGraph([(0, 1), (1, 2), (2, 1)])
+    >>> list(nx.all_neighbors(DG, 1))
+    [0, 2, 2]
+
+    Notes
+    -----
+    For undirected graphs, this function is equivalent to ``G.neighbors(node)``.
+    For directed graphs, this function returns both predecessors and successors,
+    which may include duplicates if a node is both a predecessor and successor
+    (e.g., in bidirectional edges or self-loops).
```

Comment:
These two sentences make more sense in the example section providing context for the two different examples. I.e., the first sentence can go in front of the first example, the second in front of the second.